src/bika/lims/browser/analysisrequest/add2.py

```diff
@@ -1103,51 +1105,53 @@ def ajax_get_global_settings(self):
         }
         return settings
 
+    def ajax_is_reference_value_allowed(self):
+        """Checks if the current reference value is allowed for the query
+        """
+        payload = self.get_json()
+
+        catalog = payload.get("catalog", "")
+        query = payload.get("query", {})
+        uids = payload.get("uids", [])
+        name = payload.get("name", "")
+        label = payload.get("label", "")
+        field = label or name
+
+        if all([catalog, query, uids]):
```

Comment:
This search will always be done even if the query only contains the `portal_type` and/or other criteria like `sort_on` or `sort_order`. For instance this very *naive* query will always be called, even though we expect beforehand that the result will be `{allowed: True}`, regardless of `uids`:

```python
query = {
    "portal_type": "SamplePoint",
    "is_active": True,
    "sort_on": "created",
    "sort_order": "ascending",
}
```

I suggest that if the query is made only of `portal_type`, `sort_on`, `sort_order` and `is_active`, let the function to directly return `{"allowed": True}`.

Reply:
Thanks for this idea!
Added the performance improvement in  9e0fd9176 and I think it should be trivial enough to not have too many unforeseen side-effects.